File: VB/ManualDataSet.vb
Imports System.Data
Imports System.ComponentModel

Namespace SpreadsheetImportDataTable

    Public Class ManualDataSet
        Inherits DataSet

        Public Sub New()
            MyBase.New()
            Dim table As DataTable = New DataTable("table")
            DataSetName = "ManualDataSet"
            table.Columns.Add("ID", GetType(Integer))
            table.Columns.Add("MyDateTime", GetType(Date))
            table.Columns.Add("MyRow", GetType(String))
            table.Columns.Add("MyData", GetType(Double))
            table.Constraints.Add("IDPK", table.Columns("ID"), True)
            Tables.AddRange(New DataTable() {table})
        End Sub

        Public Shared Function CreateData() As ManualDataSet
            Dim ds As ManualDataSet = New ManualDataSet()
            Dim table As DataTable = ds.Tables("table")
            table.Rows.Add(New Object() {0, Date.Today, "A", 103})
            table.Rows.Add(New Object() {1, Date.Today, "B", 200})
            table.Rows.Add(New Object() {2, Date.Today, "C", 446})
            Return ds
        End Function

'#Region "Disable Serialization for Tables and Relations"
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
        Public Overloads ReadOnly Property Tables As DataTableCollection
            Get
                Return MyBase.Tables
            End Get
        End Property

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
        Public Overloads ReadOnly Property Relations As DataRelationCollection
            Get
                Return MyBase.Relations
            End Get
        End Property
'#End Region  ' Disable Serialization for Tables and Relations
    End Class
End Namespace

File: CS/ManualDataSet.cs
using System;
using System.Data;
using System.ComponentModel;

namespace SpreadsheetImportDataTable {
    public class ManualDataSet : DataSet {
        public ManualDataSet()
            : base() {
            DataTable table = new DataTable("table");

            DataSetName = "ManualDataSet";

            table.Columns.Add("ID", typeof(Int32));
            table.Columns.Add("MyDateTime", typeof(DateTime));
            table.Columns.Add("MyRow", typeof(string));
            table.Columns.Add("MyData", typeof(double));
            table.Constraints.Add("IDPK", table.Columns["ID"], true);

            Tables.AddRange(new DataTable[] { table });
        }

        public static ManualDataSet CreateData() {
            ManualDataSet ds = new ManualDataSet();
            DataTable table = ds.Tables["table"];

            table.Rows.Add(new object[] { 0, DateTime.Today, "A", 103 });
            table.Rows.Add(new object[] { 1, DateTime.Today, "B", 200 });
            table.Rows.Add(new object[] { 2, DateTime.Today, "C", 446 });

            return ds;
        }

        #region Disable Serialization for Tables and Relations
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public new DataTableCollection Tables {
            get { return base.Tables; }
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public new DataRelationCollection Relations {
            get { return base.Relations; }
        }
        #endregion Disable Serialization for Tables and Relations
    }
}
